pkgs/by-name/on/onedrivegui/package.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, python3Packages
+, fetchFromGitHub
+, copyDesktopItems
+, onedrive
+, makeDesktopItem
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "onedrivegui";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "bpozdena";
+    repo = "OneDriveGUI";
+    rev = "v${version}";
+    hash = "sha256-HutziAzhIDYP8upNPieL2GNrxPBHUCVs09FFxdSqeBs=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+
+  propagatedBuildInputs = [ onedrive ] ++
+  builtins.attrValues { inherit (python3Packages) pyside6 requests; };
+
+  doCheck = false; # No tests defined
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "OneDriveGUI";
+      exec = "onedrivegui";
+      desktopName = "OneDriveGUI";
+      comment = "OneDrive GUI Client";
+      type = "Application";
+      icon = "OneDriveGUI";
+      terminal = false;
+      categories = [ "Utility" ];
+    })
+  ];
+
+  preBuild = ''
+    cat > setup.py << EOF
+    from setuptools import setup
+    setup(
+      name='onedrivegui',
+      version='$version',
+      scripts=[
+        'src/OneDriveGUI.py',
+      ],
+    )
+    EOF
+  '';
+
+  postInstall = ''
+    ln -s $src/src/{resources,ui} $out/bin/
+    install -Dm644 src/resources/images/OneDriveGUI.png "$out/share/icons/hicolor/48x48/apps/OneDriveGUI.png"
+    mv -v $out/bin/OneDriveGUI.py $out/bin/onedrivegui
```

Comment:
If upstream uses `OneDriveGUI.py` as the executable name, I suggest you leave it as-is and then do a symlink